### ISSUE TITLE
1.0.0 release

### ISSUE THREAD

**Issue by jasonwilliams** [toomuchdesign/next-page-tester#242]
As the functionality and API of this has started to stabilise, plus its growing a decent amount of users "7,744 downloads per week", it might be time to put out a 1.0 release, to signal this is ready to use.

I don't see the API changing significantly from this point so not much point having 0.* releases now.
Should there be breaking changes thats completely fine with major updates.

**Comment by toomuchdesign** [comment]
Hi @jasonwilliams,
I decided to keep the project in version 0 because a few parts of `next-page-tester` rely on some Next.js implementation details that - if changed - might force us to release a new major version even just to support a minor/patch release from Next.js side.

On top of this, I cannot 100% guarantee we'll be able to support all the feature we currently do if Next.js will radically change under the hood.

So, these were the reasons we I decided to stay in version 0. That said, I agree that the project has been quite stable in the last months.

My copilot here (@Meemaw), suggested to switch to standard versioning a few months ago and I'm open to consider the option 👍

**Comment by jasonwilliams** [comment]
> I decided to keep the project in version 0 because a few parts of next-page-tester rely on some Next.js implementation details that - if changed - might force us to release a new major version even just to support a minor/patch release from Next.js side.

> On top of this, I cannot 100% guarantee we'll be able to support all the feature we currently do if Next.js will radically change under the hood.

Imo I think that’s fine, that’s what major versions in semver are for. As long as the release notes are clear I don’t see anyone having an issue with that.
I know some places that are apprehensive about using 0-
version projects because it’s quite “alpha/beta-esque” and still not safe for general use, I don’t think that’s the case here. I would prefer to see multiple major version bumps than being stuck on version 0.X

Happy to hear @Meemaw’S thoughts also.

**Comment by Meemaw** [comment]
I think the API we expose is stabilised and wont change anytime soon. Having that in mind, I think we could cut `1.0.0` release. New versions of NextJS will break this library and we will have to do a few tweaks/fixes internally, but the external API should not be influenced by that.

**Comment by toomuchdesign** [comment]
Just to make sure that we are all on the same page: every internal change making the library no more backward compatible with previous Next.js versions will be a new major release for us. Meaning that we might release major versions even to just support a new Next.js minor/patch release which changes an internal we rely on. 👍

**Comment by jasonwilliams** [comment]
> Just to make sure that we are all on the same page: every internal change making the library no more backward compatible with previous Next.js versions will be a new major release for us. Meaning that we might release major versions even to just support a new Next.js minor/patch release which changes an internal we rely on. 👍

Yes that makes sense to me, as long as release notes are clear which version of next you no longer support (and why) this is fine.
